native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java

```diff
@@ -242,6 +242,12 @@ private static Object nativeBatchExecuteExecutable(BObject client, BArray paramS
                     executionResults.add(ValueCreator.createRecordValue(ModuleUtils.getModule(),
                             Constants.EXECUTION_RESULT_RECORD, resultField));
                 }
+                if (e.getMessage().contains("The statement must be executed before any results can be obtained")) {
```

Comment:
This is a wrong approach. This message can change based on drivers (even for single driver version)

Reply:
Okay.
The exception has the following properties, but we can't use the following parameter to identify the exception as those values are null or 0.
driverErrorCode
sqlServerError
SQLState
vendorCode

Reply:
1. This should be moved to JDBC module. No use parsing error messages in all connectors when we know this error wont appear.
2. Wouldn't it be much clearer to validate the configuration (default or otherwise) when jdbc module is initiated rather than parsing the error message?

Reply:
> 1. This should be moved to JDBC module. No use parsing error messages in all connectors when we know this error wont appear.

Okay, I will move

> 2. Wouldn't it be much clearer to validate the configuration (default or otherwise) when jdbc module is initiated rather than parsing the error message?

Yes. But we have to validate the URL to identify the database. (URL format: `jdbc:sqlserver://[serverName[\instanceName][:portNumber]][;property=value[;property=value]]`)

 Is it okay?

@daneshk Any suggestion about this?

Reply:
I dont think we can validate for all databases, based on the error message, seems like different types can throw different errors. For now, my suggestion would be to fix known types (mssql and oracledb). We can add the fix if this is raised for any other types